packages/ui/src/components/Designer/RightSidebar/DetailView/index.tsx

```diff
@@ -174,9 +174,10 @@ Check this document: https://pdfme.com/docs/custom-schemas`);
           validator: validateUniqueSchemaKey,
           message: i18n('validation.uniqueName'),
         }],
-        props: { autocomplete: "off" }
+        props: { autoComplete: "off" }
       },
-      required: { title: i18n('required'), type: 'boolean', span: 8, hidden: defaultSchema?.readOnly },
+      required: { title: i18n('required'), type: 'boolean', span: 8, hidden: defaultSchema?.readOnly ,disabled: '{{formData.readOnly}}',},
```

Comment:
`{{formData.readOnly}}` : It is not possible to set both ReadOnly and Required to true simultaneously.

Reply:
I think we might need a new attribute on the plugin, something like `supportsReadOnly`?  🤔 

For `multiVariableText` it handles readOnly implicitly by whether you include any `{variables}` in your text. If no variables then it's effectively readOnly, otherwise it is not.

Reply:
Hmm, I see. I understand what you're saying.

However, multiVariableText is an exception, and in most cases, it's generally easy to support read-only functionality. (I had overlooked multiVariableText's such  behavior.)

If we can avoid adding a new property by modifying multiVariableText, I think we should do that. What do you think?

Reply:
I can't think of a solution now, but I will keep thinking

Reply:
@peteward 
I tried to support read-only in the multiVariableText schema with the commit below
https://github.com/pdfme/pdfme/pull/547/commits/f7517185fd6659cd26d935eae869a45da3e3da3e

Is this modification alone insufficient?

Reply:
No, I don't think this works.

The content that you type into an MVT field determines whether it is Editable/Read Only or not. Having the tickbox doesn't make sense. But it should have a Required field.

With these changes, an Editable MVT field without variables now shows a border around in form mode:

https://github.com/user-attachments/assets/cd97c31e-a2bf-4662-8776-25a91eff4bc7

It also prevents editing in Designer mode which is a bit confusing, because we're not trying to stop the designer making changes.

https://github.com/user-attachments/assets/97d6ecfc-4d63-40d1-baf9-565be1df8b87


The correct outcome is to somehow achieve the following - no `Editable` option visible but a `Required` option:

![Screenshot 2024-08-09 at 06 03 21](https://github.com/user-attachments/assets/a40c5147-6cde-41d4-9152-dfb4abbdd605)

Reply:
I haven't tested it fully, but a simple way that seems to work:

```
      editable: { title: i18n('editable'), type: 'boolean', span: 8, hidden: defaultSchema.alwaysEditable },
      required: { title: i18n('required'), type: 'boolean', span: 16, hidden: "{{formData.editable === false}}" },
```

Then in MVT/propPanel:

```
  defaultSchema: {
    ...parentPropPanel.defaultSchema,
    type: 'multiVariableText',
    text: 'Add text here using {} for variables ',
    width: 50,
    height: 15,
    content: '{}',
    variables: [],
    alwaysEditable: true,
  },
```

Reply:
I see. I reverted the changes to the MVT schema. (https://github.com/pdfme/pdfme/pull/547/commits/f7517185fd6659cd26d935eae869a45da3e3da3e)

Instead, I made modifications in the following commit:
https://github.com/pdfme/pdfme/pull/547/commits/72c12f15607016a2773b2f97ab51921d616c3517

In other words, when readOnly in defaultSchema is explicitly set to true or false, Hiding the editable option and making it unchangeable.

What do you think about this?

![CleanShot 2024-08-09 at 20 51 15](https://github.com/user-attachments/assets/cf983d19-9a08-4425-9370-0a14b58ea7cf)

Reply:
Does `defaultSchema?.readOnly !== undefined` work for shapes which have `readOnly: true`?

Reply:
Yes, of course

![CleanShot 2024-08-10 at 08 30 11](https://github.com/user-attachments/assets/252b6325-6683-4185-b1ab-6c14c62c36d6)

Reply:
ah yes, of course 🤦 

Very clever!